sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ToStringTest.java

```diff
@@ -41,19 +46,111 @@
 
   @Test
   @Category(RunnableOnService.class)
-  public void testToStringElement() {
+  public void testToStringOf() {
     Integer[] ints = {1, 2, 3, 4, 5};
     PCollection<Integer> input = p.apply(Create.of(Arrays.asList(ints)));
-    PCollection<String> output = input.apply(ToString.<Integer>element());
-    PAssert.that(output).containsInAnyOrder(toStringList(ints));
+    PCollection<String> output = input.apply(ToString.of());
+    PAssert.that(output).containsInAnyOrder(objectToString(ints));
+    p.run();
+  }
+
+  @Test
+  @Category(RunnableOnService.class)
+  public void testToStringKV() {
+    ArrayList<KV<String, Integer>> kvs = new ArrayList<>();
+    kvs.add(KV.of("one", 1));
+    kvs.add(KV.of("two", 2));
+
+    PCollection<KV<String, Integer>> input = p.apply(Create.of(kvs));
+    PCollection<String> output = input.apply(ToString.kv());
+    PAssert.that(output).containsInAnyOrder(kvToString(kvs, ","));
+    p.run();
+  }
+
+  @Test
+  @Category(RunnableOnService.class)
+  public void testToStringKVWithDelimiter() {
+    ArrayList<KV<String, Integer>> kvs = new ArrayList<>();
+    kvs.add(KV.of("one", 1));
+    kvs.add(KV.of("two", 2));
+
+    String delimiter = "\t";
+
+    PCollection<KV<String, Integer>> input = p.apply(Create.of(kvs));
+    PCollection<String> output = input.apply(ToString.kv(delimiter));
+    PAssert.that(output).containsInAnyOrder(kvToString(kvs, delimiter));
+    p.run();
+  }
+
+  @Test
+  @Category(RunnableOnService.class)
+  public void testToStringIterable() {
+    ArrayList<Iterable<String>> iterables = new ArrayList<>();
+    iterables.add(Arrays.asList(new String[]{"one", "two", "three"}));
+    iterables.add(Arrays.asList(new String[]{"four", "five", "six"}));
+
+    PCollection<Iterable<String>> input = p.apply(Create.of(iterables)
+            .withCoder(IterableCoder.of(StringUtf8Coder.of())));
+    PCollection<String> output = input.apply(ToString.iterable());
+    PAssert.that(output).containsInAnyOrder(iterableToString(iterables, ","));
+    p.run();
+  }
+
+  @Test
+  @Category(RunnableOnService.class)
+  public void testToStringIterableWithDelimiter() {
+    ArrayList<Iterable<String>> iterables = new ArrayList<>();
+    iterables.add(Arrays.asList(new String[]{"one", "two", "three"}));
+    iterables.add(Arrays.asList(new String[]{"four", "five", "six"}));
+
+    String delimiter = "\t";
+
+    PCollection<Iterable<String>> input = p.apply(Create.of(iterables)
+            .withCoder(IterableCoder.of(StringUtf8Coder.of())));
+    PCollection<String> output = input.apply(ToString.iterable(delimiter));
+    PAssert.that(output).containsInAnyOrder(iterableToString(iterables, delimiter));
     p.run();
   }
 
-  private List<String> toStringList(Object[] ints) {
-    List<String> ll = new ArrayList<>(ints.length);
-    for (Object i : ints) {
-      ll.add(i.toString());
+  private List<String> objectToString(Object[] objects) {
```

Comment:
I'm inclined to drop all these utility methods in favor of just inlining the right answers in the tests.

Reasoning: the "right answer" is easy to see as correct on inspection, and the lines of code down (more than in the tests above!) here are not themselves covered by tests.